cognite/replicator/replication.py

```diff
@@ -146,6 +146,55 @@ def new_metadata(
     return metadata
 
 
+def restore_fields(
+    dst_obj: Union[Asset, Event, FileMetadata, TimeSeries], dst_obj_dump: Dict[str, Any], exclude_fields: List[str]
+) -> Union[Asset, Event, FileMetadata, TimeSeries]:
+    """
+    Restore the objects data according to exclude_fields defined.
+    It only support for copy timeseries.
+
+    Fields not restored
+        - **_replicatedSource**: The name of the project this object is replicated from.
+        - **_replicatedTime**: The timestamp of when the object was replicated, all objects created/updated in the same
+          execution will have the same timestamp.
+        - **_replicatedInternalId**: The internal id of the source object that the destination object
+          is being replicated from.
+
+    Args:
+        dst_obj: new object .
+        dst_obj_dump: previous object.
+        exclude_fields: List of fields:  Only support name, description, metadata and metadata.customfield.
+
+    Returns:
+        Object restored.
+
+    """
+
+    replicator_metadata_fields = ["_replicatedSource", "_replicatedTime", "_replicatedInternalId"]
+
+    for key in exclude_fields:
+        if key == "name":
+            dst_obj.name = dst_obj_dump["name"]
+        elif key == "description":
+            dst_obj.description = dst_obj_dump["description"]
+        elif key == "metadata":
+            replicator_metadata = {}
+
+            for replicator_metadata_field in replicator_metadata_fields:
```

Comment:
I'm not sure, but why do you need to loop through all fields? Can't you just set dst_obj.metadata = dst_obj_dump["metadata"] when you want to skip metadata sync?

Reply:
I think it is looping through those fields and adding them anyway

    replicator_metadata_fields = ["_replicatedSource", "_replicatedTime", "_replicatedInternalId"]

It is ok no?

Reply:
or we dont want any metadata at all, i think the replicator relies on those fields to be there